Auth Pages: user Login and user registeration pages

BrookW-83 opened the pull request:
**Summary**

> What is the reason for this Pull-Request? Are you bumping the version?


This pull request includes several changes to add authentication functionality, update UI components, and enhance the Tailwind CSS configuration. The most important changes include the addition of Google and Credentials providers for authentication, new components for registration and sign-in, and updates to the Tailwind CSS configuration for dark mode support and custom properties.

### Authentication:
* Added Google and Credentials providers for authentication in `authOptions` and created routes for handling authentication in `NextAuth`. (`app/api/auth/[...nextauth]/options.ts`, `app/api/auth/[...nextauth]/route.ts`) ([app/api/auth/[...nextauth]/options.tsR1-R54](diffhunk://#diff-256694b44e65171296996a5e26abc9085f01fad81eaabe1ec14e738dc2d8f18bR1-R54), [app/api/auth/[...nextauth]/route.tsR1-R6](diffhunk://#diff-118b33d86580f1b0da08a4a304657c7f767c63d1238634628ba80a1486575e78R1-R6))

### UI Components:
* Added `Register` component for user registration, including form handling and UI layout. (`app/components/auth/Register.tsx`)
* Added `SignIn` component for user sign-in, including form handling and UI layout. (`app/components/auth/SignIn.tsx`)


### Store and API Integration:
* Integrated authentication API with Redux Toolkit, adding `authApi` for handling registration and login requests. (`store/api/auth.ts`, `store/store.ts`) [[1]](diffhunk://#diff-8bcb149be36486cd5d455d04e4eae7d9ffcb31e3f3e325ed5d993cbd71ba8188R1-R27) [[2]](diffhunk://#diff-23f7493f4d677af7b4e76ce2e5bcd2a23e2258f36000591d12bc9ef31d5c4602R5-R16)

### Additional Changes:
* Added new dependencies in `package.json` for authentication and utility functions. (`package.json`)

BrookW-83 committed:
Implementing authenthication pages UI

BrookW-83 committed:
Implementing fetching and api calles for authentication pages

BrookW-83 committed:
Update authentication pages API calls and UI and adding responsiveness

BrookW-83 committed:
Update authentication pages API calls and UI with responsiveness

BrookW-83 committed:
chore: Update npm dependencies and add NextAuth for authentication

BrookW-83 committed:
Update Register page UI with responsive design

BrookW-83 committed:
feat: Update Register and SignIn pages with loading spinner during form submission

Added responsive desing

BrookW-83 committed:
feat: Add LoadingSpinner component for displaying loading state

BrookW-83 committed:
chore: Add CredentialsProvider for authentication

This commit adds the CredentialsProvider to the authentication options.

Joephy527 commented on store/api/auth.ts:
why do we need this api slice if we are using next auth for authentication?

Joephy527 commented on app/components/auth/SignIn.tsx:
```diff
@@ -0,0 +1,140 @@
+            <h1 className="text-Secondary font-SofiaProSemiBold p-2 md:p-4">
+              {isLoading ? (<LoadingSpinner/>) : ("Sign In")}
+            </h1>
+          </button>
```
an h1 should only be used once per page and shouldn't be inside this button

Joephy527 commented on components.json:
could you explain what the purpose of this file is?

Joephy527 reviewed (commented):
I'm not seeing where the zod validation is being enforced on the forms.

everything else is well done.

BrookW-83 committed:
fix: Addrese comments listed on PR
- Change multiple use of <h1> tags
- Remove unused package

BrookW-83 committed:
chore: Add favicon, ESSS logo

BrookW-83 committed:
fix: Add validation to input fields before submission

BrookW-83 committed:
chore: Reorganize authentication and the rest of the pages into new directory structure

- Group pages with unique layout for each group

- Fix authentication pages responsivness

- Fix responsivness for 'about pages'

BrookW-83 committed:
chore: Fix responsiveness on hero section

BrookW-83 committed:
refactor: simplify Courses page structure and update import paths

BrookW-83 committed:
fix: streamline error handling in Register and SignIn components

BrookW-83 committed:
feat: add password visibility toggle to Register and SignIn components

BrookW-83 committed:
fix: update email regex to allow all valid domain extensions
